pdq/AMC.go

```diff
@@ -0,0 +1,19 @@
+package pdq
+
+import (
+	"fmt"
+)
+
+// https://github.com/4rchr4y/go-code-metric#average-method-complexity-amc
+
+func CalcAMC(methodsComplexity int, totalMethodsNum int) (float64, error) {
+	if methodsComplexity < 0 || totalMethodsNum < 0 {
+		return 0, fmt.Errorf("method complexity and total number of methods cannot be negative, but got %d, %d", methodsComplexity, totalMethodsNum)
+	}
+
+	if methodsComplexity == 0 || totalMethodsNum == 0 {
+		return 0, nil
+	}
```

Comment:
should the result be 0 if **one** of the parameters is 0 and the other is not. For exmpl, if `methodsComplexity` is not 0, but `totalMethodsNum` is 0, this may indicate an error in the input data

Maybe could be like 

```golang
	if totalMethodsNum == 0 {
		if methodsComplexity == 0 {
			return 0, nil
		}
		return 0, fmt.Errorf("total number of methods is zero, but method complexity is %d", methodsComplexity)
	}
```